From d9239e2cee27cad582544e769ad794ca571f35fb Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <ofourdan@redhat.com>
Date: Tue, 18 May 2021 14:04:05 +0200
Subject: [PATCH] windowManager: Ignore error without systemd

On systems/setups not using systemd, a failure to start gsd-xsettings
with NOT_SUPPORTED does not denote a failure to start Xwayland, just
that we're not using systemd.

In that case, we should just ignore the error, otherwise it will prevent
Xwayland to start on such systems/setups without systemd.

Thanks to Rose Kunkel (@rosekunkel) for spotting the issue/suggesting
the fix.

Fixes: 019229c40e - windowManager: Return failure to start X11 services
Closes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/4284
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1850>
---
 js/ui/windowManager.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/js/ui/windowManager.js b/js/ui/windowManager.js
index 49d3dda229..bacb6dddb5 100644
--- a/js/ui/windowManager.js
+++ b/js/ui/windowManager.js
@@ -982,9 +982,10 @@ var WindowManager = class {
             // managed and gnome-session will have taken care of everything
             // already.
             // Note that we do log cancellation from here.
-            if (!e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.NOT_SUPPORTED))
+            if (!e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.NOT_SUPPORTED)) {
                 log('Error starting X11 services: %s'.format(e.message));
-            status = false;
+                status = false;
+            }
         } finally {
             task.return_boolean(status);
         }
-- 
GitLab

